streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -529,8 +541,7 @@ public void run() {
                 }
             }
 
-            log.error("Encountered the following exception during processing " +
-                "and the thread is going to shut down: ", e);
+            handleStreamsUncaughtException(e);
             throw e;
```

Comment:
this will trigger the old uncaught exception handler

Reply:
I'm not sure this is right, actually. If we wanted to use the current thread to send the "poison pill" subscription, it needs to keep running and call poll again.

Maybe instead we should have a default implementation of the uncaughtExceptionHandler that invokes the legacy one and then returns `SHUTDOWN_STREAM_THREAD`, and then the implementation of that case in KafkaStreams would be to re-throw the exception.

Reply:
you are right. It caused a problem with Clients with a single thread as well I moved it into run loop and this resolved the problems